### PR TITLE
SnoozeNextAd, GetAdSchedule, ad related scopes

### PR DESCRIPTION
**Pull request by swiftyspiffy**
This PR adds support for the recently added beta endpoints: Snooze Next Ad and Get Ad Schedule. It also adds support the accompanying authentication scopes.

These have not been tested yet as I need a monetized channel with ads manager enabled.

### CODE REVIEW

File: TwitchLib.Api.Helix.Models/Channels/GetAdSchedule/AdSchedule.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        public string SnoozeRefreshAt { get; protected set; }
+        /// <summary>
+        /// <para>The UTC timestamp of the broadcaster’s next scheduled ad, in RFC3339 format. Empty if the channel has no ad scheduled or is not live.</para>
+        /// </summary>
+        [JsonProperty(PropertyName = "next_ad_at")]
+        public string NextAdAt { get; protected set; }
+        /// <summary>
+        /// <para>The length in seconds of the scheduled upcoming ad break.</para>
+        /// </summary>
+        [JsonProperty(PropertyName = "length_seconds")]
```
Comment (Syzuna) [inline TwitchLib.Api.Helix.Models/Channels/GetAdSchedule/AdSchedule.cs:31]: this is actually lengths_seconds, the Twitch docs are wrong here

File: TwitchLib.Api.Helix.Models/Channels/GetAdSchedule/GetAdScheduleResponse.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+{
+    /// <summary>
+    /// <para>Response to getting ad schedule</para>
+    /// </summary>
+    public class GetAdScheduleResponse
+    {
+        /// <summary>
+        /// <para>A list that contains information related to the channel’s ad schedule.</para>
+        /// </summary>
+        [JsonProperty(PropertyName = "snooze_count")]
```
Comment (Syzuna) [inline TwitchLib.Api.Helix.Models/Channels/GetAdSchedule/GetAdScheduleResponse.cs:16]: this needs to be data

**Review by Syzuna — CHANGES_REQUESTED** [review]
LGTM but I have 2 things that need to be adjusted. 
See comments

File: TwitchLib.Api.Helix.Models/Channels/GetAdSchedule/AdSchedule.cs (hunk 1 above)
Comment (swiftyspiffy) [reply TwitchLib.Api.Helix.Models/Channels/GetAdSchedule/AdSchedule.cs:31]: iiinteresting

File: TwitchLib.Api.Helix.Models/Channels/GetAdSchedule/GetAdScheduleResponse.cs (hunk 2 above)
Comment (swiftyspiffy) [reply TwitchLib.Api.Helix.Models/Channels/GetAdSchedule/GetAdScheduleResponse.cs:16]: good catch

**Review by Syzuna — APPROVED** [review]

**Pull request merged**